runtime/near-vm-runner/src/memory.rs

```diff
@@ -25,30 +25,78 @@ impl WasmerMemory {
     }
 }
 
-impl MemoryLike for WasmerMemory {
-    fn fits_memory(&self, offset: u64, len: u64) -> bool {
-        match offset.checked_add(len) {
-            None => false,
-            Some(end) => self.0.size().bytes() >= Bytes(end as usize),
+impl WasmerMemory {
+    fn with_memory<F>(&self, offset: u64, len: usize, func: F) -> Result<(), ()>
+    where
+        F: FnOnce(core::slice::Iter<'_, std::cell::Cell<u8>>),
+    {
+        let start = usize::try_from(offset).unwrap();
+        if let Some(end) = start.checked_add(len) {
+            if Bytes(end) <= self.0.size().bytes() {
+                func(self.0.view()[start..end].iter());
```

Comment:
Can this use `.get` as well instead of checking for `end` bound manually as well?